src/deploy-cromwell-on-azure/Deployer.cs

```diff
@@ -188,32 +188,50 @@ public async Task<int> DeployAsync()
                             {
                                 retrievedKeepSshPortOpen = await GetInstalledKeepSshPortOpenAsync(sshConnectionInfo);
                             }
-
-                            var existingUserManagedIdentityId = linuxVm.UserAssignedManagedServiceIdentityIds.FirstOrDefault();
-
-                            if (existingUserManagedIdentityId == null)
-                            {
-                                managedIdentity = await ReplaceSystemManagedIdentityWithUserManagedIdentityAsync(resourceGroup, linuxVm);
-                            }
-                            else
-                            {
-                                managedIdentity = await azureSubscriptionClient.Identities.GetByIdAsync(existingUserManagedIdentityId);
-                            }
-
-                            await ConfigureVmAsync(sshConnectionInfo, managedIdentity);
                         }
                         finally
                         {
                             configuration.KeepSshPortOpen ??= retrievedKeepSshPortOpen;
                         }
 
+                        await ConfigureVmAsync(sshConnectionInfo, managedIdentity);
+
                         var accountNames = DelimitedTextToDictionary((await ExecuteCommandOnVirtualMachineWithRetriesAsync(sshConnectionInfo, $"cat {CromwellAzureRootDir}/env-01-account-names.txt || echo ''")).Output);
 
                         if (!accountNames.Any())
                         {
                             throw new ValidationException($"Could not retrieve account names from virtual machine {configuration.VmName}.");
                         }
 
+                        if (!accountNames.TryGetValue("ManagedIdentityClientId", out var existingUserManagedIdentity))
+                        {
+                            managedIdentity = await ReplaceSystemManagedIdentityWithUserManagedIdentityAsync(resourceGroup, linuxVm);
+                        }
+                        else
+                        {
+                            managedIdentity = await linuxVm.UserAssignedManagedServiceIdentityIds.Select(GetIdentityByIdAsync).FirstOrDefault(MatchesClientId);
```

Comment:
Instead of the local function with a try/catch, would a null-conditional operator handle the exception?

.FirstOrDefault(id => id.Equals(identity?.Result?.Client, StringComparison.OrdinalIgnoreCase);

Reply:
No, it wouldn't. `identity` is the task itself, so it would never be `null` (using await would have pushed the throwing of the exception to a harder-to-deal-with place). As soon as that task "goes out of scope" it will then throw if the Exception property hadn't been accessed.

I want the identity that had been configured by CoA. I don't want to error out trying to access some identity I might not have access to, and I need to "inflate" the identities in order to access their `ClientId` properties in order to find that one.